### PR TITLE
Docs: Move packages section out of CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by gziolo**
## Description
`CONTRIBUTING.md` has become quite big. This PR extracts the part about managing packages to its own file to make the guidelines more focused on the onboarding experience.

I also update internal links to follow the same patterns we started to apply recently.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -244,15 +124,15 @@ If you'd like to contribute to the design or front-end, feel free to contribute
 
 Documentation is automatically synced from master to the [Gutenberg Documentation Website](https://wordpress.org/gutenberg/handbook/) every 15 minutes.
 
-To add a new documentation page, you'll have to create a Markdown file in the [docs](https://github.com/WordPress/gutenberg/tree/master/docs) folder and add an item to the [root-manifest.json](https://github.com/WordPress/gutenberg/blob/master/docs/root-manifest.json).
+To add a new documentation page, you'll have to create a Markdown file in the [docs](https://github.com/WordPress/gutenberg/tree/master/docs) folder and add an item to the [toc.json](/docs/toc.json).
```
Comment (gziolo) [inline CONTRIBUTING.md]: Is it `toc.json` now? It has changed and I tried to figure out based on the tool which generated the manifest file.

Comment (ajitbohra) [reply CONTRIBUTING.md]: @chrisvanpatten

**Review by ajitbohra — APPROVED** [review]
LGTM :+1: 

This split will make it easy for people to refer the contribution docs 
:heart:

**Comment by mkaz** [comment]
This looks good, I have a PR that changes up the contributors handbook here: https://github.com/WordPress/gutenberg/pull/13352 but I only modified this document a little.

Go ahead and land this one and I can rebase and update.

**Review by mkaz — APPROVED** [review]

**Pull request merged**